### PR TITLE
Map View Fragment

### PR DESCRIPTION
**Pull request by ElmerAdrianV**
Implement the google maps api to show the maps and the current simulation report in the zone.

### CODE REVIEW

File: app/src/main/java/com/elmeradrianv/shesafe/MainActivity.java [hunk 1]
```diff
@@ -1,14 +1,63 @@
...
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+
+import com.elmeradrianv.shesafe.fragments.MapFragment;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
+
+import java.util.concurrent.atomic.AtomicReference;
```
Comment (sefar) [inline app/src/main/java/com/elmeradrianv/shesafe/MainActivity.java:14]: Is this needed?

File: app/src/main/java/com/elmeradrianv/shesafe/MainActivity.java [hunk 2]
```diff
@@ -1,14 +1,63 @@
...
     protected void onCreate(Bundle savedInstanceState) {
+
+
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);
+        final FragmentManager fragmentManager = getSupportFragmentManager();
+        // define your fragments here
+        final Fragment mapFragment = new MapFragment();
+//        final Fragment fragment2 = new SecondFragment();
```
Comment (sefar) [inline app/src/main/java/com/elmeradrianv/shesafe/MainActivity.java:28]: Commented out code

File: app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+            showMarker(report.getTypeOfCrime().getTag(),
+                    report.getLocation().getLatitude(),
+                    report.getLocation().getLongitude(),
+                    report.getTypeOfCrime().getLevelOfRisk());
+        }
+    }
+
+    private void showMarker(String title, double latitude, double longitude, int levelOfRisk) {
+        LatLng reportLatLng = new LatLng(latitude, longitude);
+        Marker marker;
```
Comment (sefar) [inline app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java:163]: I recommend keeping initialization of variables as close as possible to their usage.

In this case Marker marker = map.addMarker(markerOptions); would be ideal.

File: app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+                // Calculate t for bounce based on elapsed time
+                float t = Math.max(
+                        1 - interpolator.getInterpolation((float) elapsed
+                                / duration), 0);
+                // Set the anchor
+                marker.setAnchor(0.5f, 1.0f + 14 * t);
+
+                if (t > 0.0) {
+                    // Post this event again 15ms from now.
+                    handler.postDelayed(this, 15);
```
Comment (sefar) [inline app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java:193]: Might be worth considering a PinEffect class to contain the animation data.

I'd also pull the duration values (1500, 15) out into static final variables so you can see them at the top and allow for easy modification to behavior later.

File: app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+                } else { // done elapsing, show window
+                    marker.showInfoWindow();
+                }
+            }
+        });
+    }
+
+    private BitmapDescriptor getNewIcon(int levelOfRisk) {
+        BitmapDescriptor icon = BitmapDescriptorFactory.fromResource(R.drawable.ic_ss_marker_yellow);
+        ;
```
Comment (sefar) [inline app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java:203]: ?

**Review by sefar — APPROVED** [review]

File: app/src/main/java/com/elmeradrianv/shesafe/MainActivity.java (hunk 2 above)
Comment (ElmerAdrianV) [reply app/src/main/java/com/elmeradrianv/shesafe/MainActivity.java:28]: This the guide for my next code. It's more simple to me work with phantom code to follow the same pattern of work

**Pull request merged**